src/contracts/libraries/GPv2Encoding.sol

```diff
@@ -18,7 +18,7 @@ library GPv2Encoding {
         uint256 sellAmount;
         uint256 buyAmount;
         uint32 validTo;
-        uint32 appData;
+        bytes32 appData;
```

Comment:
I was wondering if this could even be variable length (in order to allow e.g. profit sharing), but I guess any bytes32 could map to a smart contract address that specifies any further referral fee logic.

Reply:
My hope is that `bytes32` should be enough. It could even be the hash of some complex data hosted on IPFS for example. The only problem I see with variable length is the extra decoding complexity.